### PR TITLE
[cli] update `dns` docs

### PR DESCRIPTION
**Pull request by kylorene**

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -3107,6 +3107,21 @@ template <> otError Interpreter::Process<Cmd("dns")>(Arg aArgs[])
...
+           Port:1234, Priority:1, Weight:2, TTL:7200
+           Host:host.example.com.
+           HostAddress:fd00:0:0:0:0:0:0:abcd TTL:7200
+           TXT:[a=1234] TTL:7300
+     * Done
+     * @endcode
+     * @code
      * dns browse _airplay._tcp.default.service.arpa
      * DNS browse response for _airplay._tcp.default.service.arpa.
      * Gabe's Mac mini
```
Comment (Vyrastas) [inline src/cli/cli.cpp:3127]: Actually, can you remove the name here, just do "Mac mini"

**Comment by Vyrastas** [comment]
After making the suggested fix please rebase upstream and squash

**Comment by Vyrastas** [comment]
@jwhui We're having a problem with the pretty check on this one, for `@cparam` lines that are too long (adding the line break that pretty wants messes up Doxygen import).  Trying to figure it out.

**Comment by Vyrastas** [comment]
@jwhui Is there a way we can bypass the make pretty check for this one, and if we do, does that still mean it's going to fail for any future update to this file?  We need to take a deeper dive into Doxygen to figure out how to handle lines that are too long for the `@cparam` tags.

**Comment by jwhui** [comment]
> @jwhui Is there a way we can bypass the make pretty check for this one, and if we do, does that still mean it's going to fail for any future update to this file? We need to take a deeper dive into Doxygen to figure out how to handle lines that are too long for the `@cparam` tags.

@Vyrastas , is it possible to wrap with `clang-format off` and `clang-format on`?

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -3025,21 +3041,22 @@ template <> otError Interpreter::Process<Cmd("dns")>(Arg aArgs[])
...
-         * [@ca{max-tx-attempts}] [@ca{recursion-desired-boolean}]
+         * @cparam dns config [@ca{dns-server-IP}] [@ca{dns-server-port}][@ca{response-timeout-ms}] [@ca{max-tx-attempts}] [@ca{recursion-desired-boolean}]
          * @par
          * We can leave some of the fields as unspecified (or use value zero). The
          * unspecified fields are replaced by the corresponding OT config option
          * definitions OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT to form the default
          * query config.
          * 'OPENTHREAD_CONFIG_DNS_CLIENT_ENABLE' is required.
          */
+        /* clang-format off */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:3052]: This one should be `clang-format on`

**Review by Vyrastas — APPROVED** [review]

**Comment by Vyrastas** [comment]
> > @jwhui Is there a way we can bypass the make pretty check for this one, and if we do, does that still mean it's going to fail for any future update to this file? We need to take a deeper dive into Doxygen to figure out how to handle lines that are too long for the `@cparam` tags.
> 
> @Vyrastas , is it possible to wrap with `clang-format off` and `clang-format on`?
> 
> https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code

This did the trick, thanks for the pointer.  I'll see if there's a way to fix our Doxygen template so we don't have to rely on this.

**Review by jwhui — APPROVED** [review]

**Pull request merged**